File: Data Structures & Algorithms/binary-tree-postorder-traversal/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    vector<int> result;

public:
    void traversal(TreeNode* node) {
        if (!node) {
            return;
        } 
        traversal(node->left);
        traversal(node->right);
        result.push_back(node->val);
    }
    
    vector<int> postorderTraversal(TreeNode* root) {
        traversal(root);
        return result;
    }
};

File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        
        maxDepth = 0

        bfs = deque([root])

        while bfs:
            for _ in range(len(bfs)):
                node = bfs.popleft()

                if node.left:
                    bfs.append(node.left)

                if node.right:
                    bfs.append(node.right)
            
            maxDepth += 1

        return maxDepth

            




        


        